syntax = "proto2";

package android.bluetooth.test_vendor_lib.model.devices.ScriptedBeaconBleAdProto;

option optimize_for = LITE_RUNTIME;

message BleAdvertisement {
  optional bytes payload = 1;
  optional bytes mac_address = 2;
  optional uint32 delay_before_send_ms = 3;
}

message BleAdvertisementList {
  repeated BleAdvertisement advertisements = 1;
}

message PlaybackEvent {
  enum PlaybackEventType {
    NOT_READY = 0;
    WAITING_FOR_FILE = 1;
    FILE_NOT_READABLE = 2;
    PLAYBACK_STARTED = 3;
    PLAYBACK_ENDED = 4;
  }
  optional PlaybackEventType type = 1;
  optional uint64 secs_since_epoch = 2;
}

message PlaybackEvents {
  repeated PlaybackEvent events = 1;
}

